tests/test_java.py

```diff
@@ -718,3 +718,42 @@ def test_java_frames_include_no_semicolons(
         assert not is_pattern_in_collapsed(r"\([^);]+;[^)]*\)", collapsed)
         # only a pipe can occur within arguments
         assert is_pattern_in_collapsed(r"\([^);|]+\|[^)]*\)", collapsed)
+
+
+# test that async profiler doesn't print anything to applications stdout, stderr streams
+@pytest.mark.parametrize("in_container", [True])
+def test_no_stray_output_in_stdout_stderr(
+    tmp_path: Path,
+    application_pid: int,
+    application_docker_container: Container,
+    monkeypatch: MonkeyPatch,
+) -> None:
+    # save original stop function
+    stop_async_profiler = AsyncProfiledProcess.stop_async_profiler
+
+    # replace async profiler stop routine to trigger flushing standard streams
+    def flush_output_and_stop_async_profiler(self: AsyncProfiledProcess, *args, **kwargs) -> str:
+        # Call 'version' and 'check' actions on async-profiler to make sure writes to stdout, stderr are flushed
+        self._run_async_profiler(
+            self._get_base_cmd() + [f"version" f",log={self._log_path_process}"],
+        )
+        self._run_async_profiler(
+            self._get_base_cmd() + [f"check" f",log={self._log_path_process}"],
+        )
+        result = stop_async_profiler(self, *args, **kwargs)
+        return result
+
+    monkeypatch.setattr(AsyncProfiledProcess, "stop_async_profiler", flush_output_and_stop_async_profiler)
+
+    with make_java_profiler(
+        storage_dir=str(tmp_path),
+        duration=3,
+        frequency=999,
+    ) as profiler:
+        snapshot_pid_profile(profiler, application_pid)
+        textout = application_docker_container.logs(stdout=True, stderr=False).decode()
+        texterr = application_docker_container.logs(stdout=False, stderr=True).decode()
+        # output from Fibonacci should be the only output in stdout
+        assert textout.splitlines() == ["Fibonacci thread starting"]
```

Comment:
```suggestion
        assert textout == "Fibonacci thread starting\n"
```

simpler? Same for stderr

Reply:
Yes.